Tools/CISupport/ews-build/steps.py

```diff
@@ -1312,9 +1339,18 @@ def validate_github(self, pr_number):
 
         repository_url = self.getProperty('repository', '')
 
-        pr_json = self.get_pr_json(pr_number, repository_url)
-        if not pr_json:
+        for index in range(self.GITHUB_RETRY + 1):
+            pr_json = self.get_pr_json(pr_number, repository_url)
+            if pr_json:
+                break
+
             self._addToLog('stdio', 'Unable to fetch pull request {}.\n'.format(pr_number))
+            if index >= self.GITHUB_RETRY:
+                break
+
+            wait_for = (index + 1) * 15
+            self._addToLog('stdio', 'Backing off for {} seconds before retrying.\n'.format(wait_for))
+            time.sleep(wait_for)
```

Comment:
Usually we don't have sleep in buildbot code. I'm little concerned about time.sleep(), if it can have any unintended consequence.

Reply:
What's our other option here, though? We either back-off to retry (with `time.sleep`), retry without a backoff or we fail the build because we can't list labels. Feels like the back-off is the least-bad option

Reply:
How often does this failure happens? Did you noticed it or just added as a precaution?
Retry without a backoff is also ok. You can also retry the build (which would take at least couple of seconds to reach this step again)

Reply:
It's relatively common, I've seen it in production a number of times, but it's not a fatal failure for most queues so it's not a big deal.

The real kicker is that for `Merge-Queue`, failing to retrieve the labels _is_ fatal, and we have to check between each step to abort merging if someone has either removed the merge label or blocked the change. So "just retry the build" is not an option.